server/channels/app/user.go

```diff
@@ -106,7 +108,9 @@ func (a *App) CreateUserWithToken(c request.CTX, user *model.User, token *model.
 		return nil, err
 	}
 
-	a.AddDirectChannels(c, team.Id, ruser)
+	if appErr := a.AddDirectChannels(c, team.Id, ruser); appErr != nil {
+		c.Logger().Warn("Error while adding direct channels", mlog.Err(appErr))
+	}
```

Comment:
We return an error if `JoinUserToTeam` fails. I think it's fine to also error here.